File: ios/Plugin/ChangeAppIconPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(ChangeAppIconPlugin)
public class ChangeAppIconPlugin: CAPPlugin {
    private let implementation = ChangeAppIcon()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }

    @objc func changeAppIcon(_ call: CAPPluginCall) {
        let name = call.getString("name")
        print("icon name change to:\(name)")
        implementation.changeAppIcon(name, call);
        
    }
    
}

File: android/src/main/java/com/aigens/change/app/icon/ChangeAppIconPlugin.java
package com.aigens.change.app.icon;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.NativePlugin;


@NativePlugin()
public class ChangeAppIconPlugin extends Plugin {

    private ChangeAppIcon implementation = new ChangeAppIcon();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");
        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void changeAppIcon(PluginCall call) {
        implementation.changeAppIcon(call, this.getActivity().getBaseContext(), this.getActivity().getPackageManager());
    }
}
